packages/amplify-codegen-e2e-tests/src/__tests__/codegen-matrix.test.ts

```diff
@@ -0,0 +1,220 @@
+import { DEFAULT_JS_CONFIG, createNewProjectDir, initProjectWithProfile, createRandomName, addApiWithoutSchema, updateApiSchema, apiGqlCompile, removeCodegen, amplifyConfigureProjectInfo, AmplifyFrontend, DEFAULT_FLUTTER_CONFIG, DEFAULT_IOS_CONFIG, DEFAULT_ANDROID_CONFIG } from "@aws-amplify/amplify-codegen-e2e-core";
+import { CodegenMatrixTestProps, deleteAmplifyProject, testAddCodegenMatrix } from "../codegen-tests-base";
+
+const schema = 'simple_model.graphql';
+
+describe('JS codegen matrix test', () => {
+  let projectRoot: string;
+  let config: any = DEFAULT_JS_CONFIG;
+
+  beforeAll(async () => {
+    projectRoot = await createNewProjectDir('addCodegenMatrixJS');
+    // init project and add API category
+    await initProjectWithProfile(projectRoot, { ...config });
+    const projectName = createRandomName();
+    await addApiWithoutSchema(projectRoot, { apiName: projectName });
+    await updateApiSchema(projectRoot, projectName, schema);
+    await apiGqlCompile(projectRoot);
+  });
+  beforeEach(async () => {
+
+  });
+  afterEach(async () => {
+    await removeCodegen(projectRoot);
+    await amplifyConfigureProjectInfo({ cwd: projectRoot, frontendType: 'flutter'} )
```

Comment:
This is to reset the project to the flutter frontend type. This is because of the helper function `singleSelect` method cannot be executed more than once when the previous selection is set as the default starting option. The typical scenario is for choosing the framework type for JS app. A reset of `flutter` app will always let the `framework` options starting at `none` option